lang/funcs/collection.go

```diff
@@ -106,6 +107,145 @@ var LengthFunc = function.New(&function.Spec{
 	},
 })
 
+// CoalesceListFunc contructs a function that takes any number of list arguments
+// and returns the first one that isn't empty.
+var CoalesceListFunc = function.New(&function.Spec{
+	Params: []function.Parameter{},
+	VarParam: &function.Parameter{
+		Name:             "vals",
+		Type:             cty.DynamicPseudoType,
+		AllowUnknown:     true,
+		AllowDynamicType: true,
+		AllowNull:        true,
+	},
+	Type: func(args []cty.Value) (ret cty.Type, err error) {
+		if len(args) == 0 {
+			return cty.NilType, fmt.Errorf("at least one argument is required")
+		}
+
+		argTypes := make([]cty.Type, len(args))
+
+		for i, arg := range args {
+			argTypes[i] = arg.Type()
+		}
+
+		retType, _ := convert.UnifyUnsafe(argTypes)
+		if retType == cty.NilType {
+			return cty.NilType, fmt.Errorf("all arguments must have the same type")
+		}
+
+		return retType, nil
+	},
+	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+
+		vals := make([]cty.Value, 0, len(args))
+		for _, arg := range args {
+
+			// We already know this will succeed because of the checks in our Type func above
+			arg, _ = convert.Convert(arg, retType)
+
+			it := arg.ElementIterator()
+			for it.Next() {
+				_, v := it.Element()
+				vals = append(vals, v)
+			}
+
+			if len(vals) > 0 {
+				return cty.ListVal(vals), nil
+			}
+		}
+
+		return cty.NilVal, fmt.Errorf("no non-null arguments")
+	},
+})
+
+// CompactFunc contructs a function that takes a list of strings and returns a new list
+// with any empty string elements removed.
+var CompactFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "list",
+			Type: cty.List(cty.String),
+		},
+	},
+	Type: function.StaticReturnType(cty.List(cty.String)),
+	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+
+		var outputList []cty.Value
+
+		for it := args[0].ElementIterator(); it.Next(); {
+			_, v := it.Element()
+			if v.AsString() == "" {
+				continue
+			}
+			outputList = append(outputList, v)
+		}
+		return cty.ListVal(outputList), nil
+	},
+})
+
+// ContainsFunc contructs a function that determines whether a given list contains
+// a given single value as one of its elements.
+var ContainsFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "list",
+			Type: cty.List(cty.DynamicPseudoType),
+		},
+		{
+			Name: "value",
+			Type: cty.DynamicPseudoType,
+		},
+	},
+	Type: function.StaticReturnType(cty.Bool),
+	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+
+		_, err = Index(args[0], args[1])
+		if err != nil {
+			fmt.Println(err)
```

Comment:
I think you left some debug code behind here.

Reply:
I'm just glad it wasn't `fmt.Println("hi mom!")`. 'Cuz I do that a _lot_.